pkg/gce-pd-csi-driver/node.go

```diff
@@ -365,7 +368,72 @@ func (ns *GCENodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGe
 }
 
 func (ns *GCENodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, fmt.Sprintf("NodeExpandVolume is not yet implemented"))
+	volumeID := req.GetVolumeId()
+	if len(volumeID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ControllerExpandVolume volume ID must be provided")
+	}
+	capacityRange := req.GetCapacityRange()
+	reqBytes, err := getRequestCapacity(capacityRange)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("ControllerExpandVolume capacity range is invalid: %v", err))
+	}
+
+	volumePath := req.GetVolumePath()
+	if len(volumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ControllerExpandVolume volume path must be provided")
+	}
+
+	volKey, err := common.VolumeIDToKey(volumeID)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("ControllerExpandVolume volume ID is invalid: %v", err))
+	}
+
+	devicePath, err := ns.getDevicePath(volumeID, "")
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("ControllerExpandVolume error when getting device path for %s: %v", volumeID, err))
+	}
+
+	// TODO(#328): Use requested size in resize if provided
+	resizer := resizefs.NewResizeFs(ns.Mounter)
+	_, err = resizer.Resize(devicePath, volumePath)
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("ControllerExpandVolume error when resizing volume %s: %v", volKey.String(), err))
+
+	}
+
+	// Check the block size
+	gotBlockSizeBytes, err := ns.getBlockSizeBytes(devicePath)
+	if err != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("ControllerExpandVolume error when getting size of block volume at path %s: %v", devicePath, err))
+	}
+	if gotBlockSizeBytes != reqBytes {
```

Comment:
Maybe this should be got >= req?  I believe GCE rounds up (?) to the nearest Gb?

Reply:
I am not sure if we need to check disk size at all here.

Reply:
think im going to continue to verify disk size but just check for `got >= req`. Just as a quick sanity check. It's not super necessary but might catch errors if for some reason the disk is not resized properly